Fabric.Identity.API/scripts/tests/CatalystDosIdentity.functional.tests.ps1

```diff
@@ -106,15 +105,151 @@ Describe 'Identity Cli Functional Tests' {
-    
-    Describe 'New-ApiRegistration' {}
-    
-    Describe 'Invoke-UpdateApiRegistration' {}
-    
-    Describe 'Invoke-UpdateClientPassword' {}
-    
-    Describe 'Test-IsApiRegistered' {}
 
+    Describe 'Api Registration Functional Tests' -tag "Functional" {
+	    Context 'Api Registration Scenarios' {
+            	# Get Access token from identity to pass to registration
+			    $accessToken = Get-AccessToken -identityUrl $identityUrl -clientId "fabric-installer" -secret $installerSecret -scope "fabric/identity.manageresources"
+
+                # Get-FabricInstallerAccessToken returns an access token if request succeeds, expect a value when successful
+                $accessToken | Should -Not -Be $null
+
+                # New-ApiRegistrationBody returns an apiresource object
+                $newApiResource = New-ApiRegistrationBody `
+                -apiName "test-Api" `
+                -scopes @{"name" = "test-Api"; "displayName" = "Test-API"} `
+                -userClaims @("name", "email", "role", "groups") `
+                -isEnabled true
+
+                $jsonApi = $newApiResource | ConvertTo-Json
+
+            It 'Should not fail to Test, Create, Test, Get, Edit and Delete Registration' {
+
+                # Test an api not registered
+                $testApi = Test-IsApiRegistered -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $testApi | Should -Be $false
+
+                # Create an api not registered
+                $newApi = New-ApiRegistration -identityUrl $identityUrl -body $jsonApi -accessToken $accessToken
+
+                $newApi | Should -Not -Be $null
+
+                # Test an api that is registered
+                $testApi = Test-IsApiRegistered -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $testApi | Should -Be $true
+
+                # Get an api that is registered
+                $getApi = Get-ApiRegistration -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $getApi | Should -Not -Be $null
+                $getApi.name | Should -Be "test-Api"
+
+                # Edit an api that is registered
+                $newApiResource = New-ApiRegistrationBody `
+                -apiName "test-Api" `
+                -scopes @{"name" = "patient-Api"; "displayName" = "Patient-API"} `
+                -userClaims @("name", "email", "role", "groups") `
+                -isEnabled true
+
+                $jsonApi = $newApiResource | ConvertTo-Json
+
+                $editApi = Edit-ApiRegistration -identityUrl $identityUrl -body $jsonApi -apiName "test-Api" -accessToken $accessToken
+
+                $editApi | Should -Not -Be $null
+
+                # Cleanup the registered Api
+                $removeApi = Remove-ApiRegistration -identityUrl $identityUrl -apiName "test-Api" -accessToken $accessToken
+
+                $removeApi | Should -Be ""
```

Comment:
To really make this test effective, should we try a get on the removed api and ensure we get a 404?

Reply:
I have another test in this group that does a Get, Edit and Remove on a non-existent api that return not found.